pymeasure/instruments/pendulum/cnt91.py

```diff
@@ -0,0 +1,182 @@
+    strict_range,
+    truncated_range,
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class CNT91(Instrument):
+    """Represents a Pendulum CNT-91 frequency counter."""
+
+    CHANNELS = {"A": 1, "B": 2, "C": 3, "E": 4, "INTREF": 6}
+    MAX_BUFFER_SIZE = 32000  # User Manual 8-38
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(resourceName, "Pendulum CNT-91", **kwargs)
+        self.adapter.connection.timeout = 120000
+        self.adapter.connection.read_termination = "\n"
+
+    @property
+    def batch_size(self):
+        """Maximum number of buffer entries that can be transmitted at once."""
+        if not hasattr(self, "_batch_size"):
+            self._batch_size = int(self.ask("FORM:SMAX?"))
+        return self._batch_size
+
+    def read_buffer(self, expected_length=0):
+        """
+        Read out the entire buffer.
+
+        :param expected_length: The expected length of the buffer. If more
+                                data is read, values at the end are be removed.
+                                Defaults to 0, which means that the entire
+                                buffer is returned independent of its length.
+        :return: Frequency values from the buffer.
+        """
+        while not self.complete:
+            # Wait until the buffer is filled.
+            sleep(0.01)
+        data = []
+        # Loop until the buffer is completely read out.
+        while True:
+            # Get maximum number of buffer values.
+            new = self.values(":FETC:ARR? MAX")
+            data += new
+            # Last values have been read from buffer.
+            if len(new) < self.batch_size:
+                # Remove the last values if the buffer is too long.
+                if expected_length and len(data) > expected_length:
+                    data = data[:expected_length]
+                    log.info("Buffer was too long, truncated.")
+                break
+        return data
+
+    external_start_arming_source = Instrument.control(
+        "ARM:SOUR?",
+        "ARM:SOUR %s",
+        """
+        Select arming input or switch off the start arming function.
+        Options are 'A', 'B' and 'E' (rear). 'IMM' turns trigger off.
+        """,
+        validator=strict_discrete_set,
+        values={"A": "EXT1", "B": "EXT2", "E": "EXT4", "IMM": "IMM"},
+        map_values=True,
+    )
+
+    external_arming_start_slope = Instrument.control(
+        "ARM:SLOP?",
+        "ARM:SLOP %s",
+        "Set slope for the start arming condition.",
+        validator=strict_discrete_set,
+        values=["POS", "NEG"],
+    )
+
+    continuous = Instrument.control(
+        "INIT:CONT?",
+        "INIT:CONT %s",
+        "Controls whether to perform continuous measurements.",
+        strict_discrete_set,
+        values=[True, False],
+        set_process=lambda x: "ON" if x else "OFF",
+        get_process=lambda x: bool(x),
+    )
+
```

Comment:
What about using
`values = {True: "1", False : "0"}` and `map_values = True` ?

Reply:
Yes, much cleaner. I thought for some reason that booleans are not allowed as dictionary keys. Also just realized that the mapping works in both directions.

Fixed in 9885993.